corehq/apps/reports/standard/deployments.py

```diff
@@ -311,7 +313,8 @@ def rows(self):
                 user_dim__user_id__in=users
             ).order_by(sort_clause).select_related('user_dim', 'app_dim')
             if self.selected_app_id:
-                rows.filter(app_dim__app_id=self.selected_app_id)
+                rows = rows.filter(app_dim__application_id=self.selected_app_id)
+            self._total_records = len(rows)
```

Comment:
Won't this fetch all the records? I think a `.count()` might be better

Reply:
my understanding was that using `len` here only fetches the rows once, and then they are used in the function call below. whereas we would still fetch all the rows below otherwise, but add an addition `COUNT` query if we use `.count()` here. i thought the rule of thumb was `.count()` if thats all you need or `len` if you were using the results otherwise. https://docs.djangoproject.com/en/2.0/ref/models/querysets/#django.db.models.query.QuerySet.count

Reply:
But you aren't using all the rows, you're slicing the result set based on the pagination.

Reply:
that is a very good point. i will fix it